README.md

```diff
@@ -100,25 +133,29 @@ sequenceDiagram
     end
     Tier 1 ->> Tier 2: Code executed more <br> than 63 times and <br> Tier 1 instructions <br> present, triggers the <br> Tier 2 interpreter
     loop Until exit scope executed
-        loop until Tier 2 encounters type-specialised tier 1 instruction
+        loop until Tier 2 encounters type-specialised tier 1 instruction or a branch instruction
             Note over Tier 2: Tier 2 copies Tier 1 <br> instructions into a <br> buffer to be executed <br> according to runtime <br> type info
             Tier 2 ->> Type Propagator: Requests type propagator 
             Type Propagator ->> Tier 2: Type propagator <br> updates runtime type <br> info based on <br>newly emitted code
         end
-        Note over Tier 2: Emits a typeguard <br> and executes Tier 2 code <br> until typeguard is hit.
+        Note over Tier 2: Emits a typeguard <br> or branch
+        Note over Tier 2: JIT compiles current <br> basic block into <br> machine code
+        Note over Tier 2: Execute machine code <br> until the end of the <br> basic block
+        Note over Tier 2: Execute the Tier 2 <br> branch instruction <br> or type guard
         Tier 2 ->> Type Propagator: Requests type propagator
         Type Propagator ->> Tier 2: Type propagator updates <br> runtime type info <br> based on  branch taken
-        Note over Tier 2: Emits type specialised <br> branch according to <br> runtime type info
+        Note over Tier 2: Starts emitting <br> type specialised <br> branch according to <br> runtime type info
     end
 ```
 
 ## What's left for our project
 
-- The Copy and Patch JIT compiler.
+- Just general bugfixing.
+- Evaluation and benchmarking
 
 ## Evaluation and benchmarks
 
-We will run the [bm_nbody.py](./bm_nbody.py) script and the [bm_float_unboxed.py](./bm_float_unboxed.py) to gather results. For now we expect performance to have no improvement as we have yet to implement the copy and patch JIT compiler.
+We will run the [bm_nbody.py](./bm_nbody.py) script and the [bm_float_unboxed.py](./bm_float_unboxed.py) to gather results. We expect at least some performance improvements now that we have implemented the copy and patch JIT compiler.
```

Comment:
No clue how it will perform until benchmarks run, so removing this line
```suggestion
We will run the [bm_nbody.py](./bm_nbody.py) script and the [bm_float_unboxed.py](./bm_float_unboxed.py) to gather results.
```